lib/src/util/console_log_utils.dart

```diff
@@ -152,3 +154,51 @@ class ConsoleConfig {
   /// Captures calls to `console.log`, `console.warn`, and `console.error`.
   static const ConsoleConfig all = ConsoleConfig._('all');
 }
+
+final formatRegExp = RegExp('%[sdj%]');
+String format(dynamic f, List<dynamic> arguments) {
+  if (f is! String) {
+    final objects = [];
+    for (var i = 0; i < arguments.length; i++) {
+      objects.add(arguments[i]);
+    }
+    return objects.join(' ');
+  }
+
+  var str = '';
+  if (f is String) {
+    if (!f.contains('%')) return f;
+
+    var i = 0;
+    final args = arguments;
```

Comment:
#nit this variable is unnecessary; why not just use `arguments`?

Reply:
i legit copy pasted and converted the node js impl i have no idea.

Reply:
Can you please attribute that code then? 😛